### PR TITLE
Collect compaction stats by priority and dump to info LOG

### PR DESCRIPTION
**Pull request by miasantreble**
In order to better understand compaction done by different priority thread pool, we now collect compaction stats by priority and also print them to info LOG through stats dump. 

```
** Compaction Stats [default] **
Priority    Files   Size     Score Read(GB)  Rn(GB) Rnp1(GB) Write(GB) Wnew(GB) Moved(GB) W-Amp Rd(MB/s) Wr(MB/s) Comp(sec) CompMergeCPU(sec) Comp(cnt) Avg(sec) KeyIn KeyDrop
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Low      0/0    0.00 KB   0.0     16.8    11.3      5.5       5.6      0.1       0.0   0.0    406.4    136.1     42.24             34.96        45    0.939     13M  8865K
High      0/0    0.00 KB   0.0      0.0     0.0      0.0      11.4     11.4       0.0   0.0      0.0     76.2    153.00             35.74     12185    0.013       0      0
```

### CODE REVIEW

File: db/db_impl_compaction_flush.cc [hunk 1]
```diff
@@ -1943,7 +1947,7 @@ void DBImpl::SchedulePendingPurge(std::string fname, std::string dir_to_sync,
 void DBImpl::BGWorkFlush(void* db) {
   IOSTATS_SET_THREAD_POOL_ID(Env::Priority::HIGH);
   TEST_SYNC_POINT("DBImpl::BGWorkFlush");
-  reinterpret_cast<DBImpl*>(db)->BackgroundCallFlush();
+  reinterpret_cast<DBImpl*>(db)->BackgroundCallFlush(Env::Priority::HIGH);
```
Comment (riversand963) [inline db/db_impl_compaction_flush.cc]: Flush can also be scheduled onto a thread in the low-priority pool. See https://github.com/facebook/rocksdb/blob/master/db/db_impl_compaction_flush.cc#L1791

File: db/db_impl_compaction_flush.cc [hunk 2]
```diff
@@ -2146,7 +2151,7 @@ void DBImpl::BackgroundCallFlush() {
 }
 
 void DBImpl::BackgroundCallCompaction(PrepickedCompaction* prepicked_compaction,
-                                      Env::Priority bg_thread_pri) {
+                                      Env::Priority thread_pri) {
```
Comment (riversand963) [inline db/db_impl_compaction_flush.cc]: Any reason for the renaming from `bg_thread_pri` to `thread_pri`?

File: db/db_impl_compaction_flush.cc (hunk 1 above)
Comment (riversand963) [reply db/db_impl_compaction_flush.cc]: You may need to add a new member variable to `BGFlushArg` and set it in `MaybeScheduleFlushOrCompaction`. Then `BackgroundCallFlush` can read it while popping elements from the queue.

File: util/threadpool_imp.cc [hunk 3]
```diff
@@ -292,6 +292,7 @@ void* ThreadPoolImpl::Impl::BGThreadWrapper(void* arg) {
     case Env::Priority::BOTTOM:
       thread_type = ThreadStatus::BOTTOM_PRIORITY;
       break;
+    case Env::Priority::USER:
```
Comment (riversand963) [inline util/threadpool_imp.cc]: Is fall-through intended?

Comment (miasantreble) [reply util/threadpool_imp.cc]: I think it should be 
```
case Env::Priority::USER:
  thread_type = ThreadStatus::USER;
  break;
```

File: db/db_impl_compaction_flush.cc [hunk 4]
```diff
@@ -1987,11 +2002,17 @@ void DBImpl::UnscheduleCallback(void* arg) {
     }
     delete ca.prepicked_compaction;
   }
-  TEST_SYNC_POINT("DBImpl::UnscheduleCallback");
+  TEST_SYNC_POINT("DBImpl::UnscheduleCompactionCallback");
+}
+
+void DBImpl::UnscheduleFlushCallback(void* arg) {
+  delete reinterpret_cast<FlushThreadArg*>(arg);
+  TEST_SYNC_POINT("DBImpl::UnscheduleFlushCallback");
```
Comment (riversand963) [inline db/db_impl_compaction_flush.cc]: Is this sync point used?

Comment (miasantreble) [reply db/db_impl_compaction_flush.cc]: Neither are used actually

**Pull request closed without merging**